keras_nlp/models/gpt_neox/gpt_neox_decoder.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.models.gpt_neox.gpt_neox_attention import GPTNeoXAttention
+from keras_nlp.utils.keras_utils import clone_initializer
+
+from keras_nlp.layers.transformer_layer_utils import (  # isort:skip
+    compute_causal_mask,
+    merge_padding_and_attention_mask,
+)
+
+
+class GPTNeoXDecoder(keras.layers.Layer):
+    def __init__(
+            self,
+            intermediate_dim,
+            num_heads,
+            max_position_embeddings=512,
+            dropout=0.,
+            activation="relu",
+            layer_norm_epsilon=1e-5,
+            rotary_pct=0.25,
+            rotary_emb_base=10000,
+            kernel_initializer="glorot_uniform",
+            bias_initializer="zeros",
+            use_parallel_residual=True,
```

Comment:
is this value both true and false for checkpoints we care about? or does one win out?

if the latter, we could consider ditching this

Reply:
Yes it is `True` for all pythia and gpt-neox-20b checkpoints